src/Board/Board.ts

```diff
@@ -876,6 +876,86 @@ module jsflap.Board {
             };
         }
 
+        public toJSON(): string {
```

Comment:
Ok, this is in the right position compared to the other changes, however it is not valid JSON. Please refactor this into just setting values in a plain JavaScript object and then call `JSON.stringify` on the resulting object. You shouldn't have to do any string manipulation and the code will be *a lot* easier to read/write.

Also just a nit, lowercase dictionary keys :)